### ISSUE TITLE
Questions about Databases.

### ISSUE THREAD

**Issue by lasers** [libretro/libretro-database#302]
Questions about databases. Are RetroArch going to get them from somewhere periodically or will RetroArch maintain the databases? I'm sure they initially got it from somewhere.

I asked because the database contains lot of various names or abbreviations for same thing... or to include extra symbols. I'm sure it is like this for all other databases too. For example... When I'm grepping `Sega - Saturn.dat` with case insensitive with `wc -l` for line numbers, I get this...

`(JP)` 1124
`(Japan)` 1078
`(JAP)` 357

`(EU)` 242
`(Eur)` 193
`(Europe)` 247

`(US)` 338
`(USA)` 467

`(NTSC)` 314
`(PAL)` 229

There are also some extra things like... 
`"Yuushun Classic Road (JP) - &#20778;&#39423;&#12463;&#12521;&#12471;&#12483;&#12463;&#12525;&#12540;&#12489;"`

I'm guessing it was a result of something or is it to display Japan letters? Or is it garbage?

I asked this because there are lot of same games with different abbreviations like this...
```
name "Z (EU)"
rom  "Z (EU).cue
name "Z (EUR) (DW0506)"
rom  "Z (EUR) (DW0506).cue
name "Z (Europe)"
rom  "Z (Europe).cue

... all with different checksum.
```
Some name, desc, or rom contain `~` as parenthesis. Easily confusing those files as a backup files.
```
name "X-Men ~Children of the Atom~ (JP) - &#12456;&#12483;...
description "X-Men ~Children of the Atom~ (JP) - &#12456;&#12483;...
rom ( name "X-Men ~Children of the Atom~ (JP).cue" size 12436 crc 3a2cc5a1... 
```
```
name "Hexen ~Beyond Heretic~ (US)"
rom  "Hexen ~Beyond Heretic~ (US).cue
```
It should be...
```
name "Hexen - Beyond Heretic (US)"
rom  "Hexen - Beyond Heretic (US).cue
```

Not digging fun. I'm just trying to understand how to combat with this... or if the databases were simply merged. At some point, I was considering using `(JP)`,`(JAP)`, or `(Japan)` in cheat codes too for exclusive Japan games. If I were to do that... Why not typing `(US)`, `(EU)`, `(BR)`, `(KR)`, `(CH)` for all games too?

`Alien Trilogy (US, EU, JP).cht` 

If we were to follow no-import style, I gather it would be something like this...

`Alien Trilogy (USA, Europe, Japan).cht` 

What would work well? Multiple regions/merged cheats for a same game in a single file?

I think lot of databases were merged to make it easy for people to scan games and add it to their list... So  that's that... even with odd or missing names.
```
name "Hexen - Beyond Heretic (Japan)"
rom  "Hexen - Beyond Heretic (Japan).cue
name "Hexen (EU)"
rom  "Hexen (EU).cue
name "Hexen (EUR) (DW0164)"
rom  "Hexen (EUR) (DW0164).cue
name "Hexen (Europe) (En,Fr,De,Es,It)"
rom  "Hexen (Europe) (En,Fr,De,Es,It).cue
name "Hexen (USA)"
rom  "Hexen (USA).cue
name "Hexen (USA) (DW0750)"
rom  "Hexen (USA) (DW0750).cue
name "Hexen ~Beyond Heretic~ (US)"
rom  "Hexen ~Beyond Heretic~ (US).cue
name "Hexen v1.002 (1997)(GT Interactive)(PAL)[!]"
rom  "Hexen v1.002 (1997)(GT Interactive)(PAL)[!].cue
... all with different checksum.
```
I'm asking all this to see if it's worth making a dedicated cheating database. Right now, I don't know if there is such a database like that and it seems like most cheat codes are just from multiple forums, multiple cheating sites, multiple guides, etc.. but no dedicated  clean database where people can make sure everything is in order, working order, naming style, etc...

tl;dr -- Cheating sucks.

**Comment by RobLoach** [comment]
I'd like to problematically standardize some of the names coming in from [libretro-dats](http://github.com/robloach/libretro-dats) (via Redump, TOSEC, etc). Right now, they're just coming in 1:1.

**Comment by lasers** [comment]
I guess I should start somewhere in `/metadat/no-intro` instead? What would you like to do with this information?
Original File: `GLoc Air Battle (Action Replay).cht`
```
chris@z420 ~ $ dat2grep -nrdi g-loc 
n G-LOC - Air Battle (Japan)
d G-LOC - Air Battle (Japan)
r G-LOC - Air Battle (Japan).gg
n G-LOC - Air Battle (Japan) (Rev 1)
d G-LOC - Air Battle (Japan) (Rev 1)
r G-LOC - Air Battle (Japan) (Rev 1).gg
n G-LOC - Air Battle (USA, Europe)
d G-LOC - Air Battle (USA, Europe)
r G-LOC - Air Battle (USA, Europe).gg
```
New: `G-LOC - Air Battle (?????) (Action Replay).cht`

Which should I name it?

* `G-LOC - Air Battle (Japan) (Action Replay).cht`
* `G-LOC - Air Battle (USA, Europe) (Action Replay).cht`
* `G-LOC - Air Battle (USA, Europe, Japan) (Action Replay).cht`

I'm leaning on `No. 3` as I don't care to test them all, but knowing that the cheating code is there for everyone to use... but that would result in long (but nice) names for all cheating codes.... just like the games! 

If we're going to use long names, we should get the team to turn off compact / alignment mode in cheating menu. It's toggled on forever... for alignment, my wild guess. I will try my best using no-intro naming style. :+1:

**Comment by lasers** [comment]
Also, what about characters? 
Quotes? `George Foreman's KO Boxing (USA, Europe)` 
Comma? `George Foreman's KO Boxing (USA, Europe)`

Do you support symbolic links? Would RetroArch download them as individual files? 
Two same games with different titles... or different games (hacks/mod) that origins from one particular game... or company offering same codes for many of their different games? 

Don't worry. I'm already renaming things.
